src/main/java/teammates/storage/entity/CourseStudent.java

```diff
@@ -20,7 +20,14 @@
  * [enrolled in] --> Course.
  */
 @PersistenceCapable
-public class CourseStudent implements StoreCallback {
+public class CourseStudent extends Entity implements StoreCallback {
+
+    /**
+     * The name of the primary key of this entity type.
+     */
+    @NotPersistent
+    public static final String PRIMARY_KEY_NAME = getFieldWithPrimaryKeyAnnotation(CourseStudent.class);
+
```

Comment:
These changes don't affect the actual entities persisted to/ read from the datastore - only a plain method ([ignored by JDO](https://cloud.google.com/appengine/docs/standard/java/datastore/jdo/dataclasses#Class_and_Field_Annotations)) and a `@NotPersistent` field are added. The new abstract class `Entity` is not `@PersistenceCapable` and does not affect the entities - it is only used to add the method. So interaction with legacy data is not affected.

However I do acknowledge that the only way to be sure would be to run tests against legacy data 😅